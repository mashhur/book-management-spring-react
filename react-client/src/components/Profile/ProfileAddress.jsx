import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import AddressService from "../../service/AddressService";
import AddressItem from "./AddressItem/AddressItem";

const Profileaddress = () => {
	const user = useSelector((state) => state.user.user);
	const [componentState, setComponentState] =
		React.useState({
			isLoading: true,
			error: "",
			addresses: [],
		});

	const { addresses, isLoading, error } = componentState;

	React.useEffect(() => {
		AddressService.getAddressByUser(user.cid)
			.then((res) => {
				setComponentState({
					...componentState,
					isLoading: false,
					addresses: res.data,
				});
			})
			.catch((err) => {
				setComponentState({
					...componentState,
					isLoading: false,
					error: err.message,
				});
			});
	}, []);

	return (
		<div
			data-test="component-profile-address"
			className=" container ">
			<div>
				<h4>Saved address</h4>
				<p>
					The following addresses are saved for you. You can
					use them to ship your books.
				</p>
				{isLoading ? (
					<div
						data-test="loading-spinner"
						className="d-flex justify-content-center">
						<div
							className="spinner-border"
							role="status"></div>
					</div>
				) : error ? (
					<div
						data-test="error-message"
						className="text-center text-danger">
						{error}
					</div>
				) : (
					<div className="list-group gy-2 mt-4">
						{addresses.map((address) => (
							<AddressItem
								key={address.aid}
								address={address}
							/>
						))}
					</div>
				)}
			</div>
		</div>
	);
};

export default Profileaddress;
