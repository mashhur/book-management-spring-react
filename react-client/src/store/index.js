import { configureStore } from "@reduxjs/toolkit";
import locationReducer from "./slices/locationSlice";
import userReducer from "./slices/userSlice";

export const store = configureStore({
	reducer: {
		location: locationReducer,
		user: userReducer,
	},
});
