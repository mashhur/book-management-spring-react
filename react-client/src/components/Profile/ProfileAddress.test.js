import React from "react";
import { mount, shallow } from "enzyme";
import { findByTestAttr } from "../../tests/testUtils";

import ProfileAddress from "./ProfileAddress";
import { Provider } from "react-redux";
import { createTestStore } from "../../store";

const setup = (props = {}) => {
	const store = createTestStore();
	return mount(
		<Provider store={store}>
			<ProfileAddress />
		</Provider>
	);
};

describe("Profile orders [parent] component test casses", () => {
	it("renders without any error", async () => {
		const wrapper = setup();
		const component = await findByTestAttr(
			wrapper,
			"component-profile-address"
		);
		expect(component.length).toBe(1);
	});

	it("calls the `getAllOrdersFor` method", async () => {
		try {
			jest
				.spyOn(React, "useEffect")
				.mockImplementation((f) => f());
			jest.spyOn(OrderService, "getAllOrdersFor");
			setup();
			await expect(
				OrderService.getAllOrdersFor
			).toHaveBeenCalledTimes(1);
		} catch (err) {}
	});

	// it("should render all the available orders", async () => {
	// 	const wrapper = setup({
	// 		orders: multipleOrders,
	// 	});
	// 	const component = await findByTestAttr(
	// 		wrapper,
	// 		"order-item-component"
	// 	);
	// 	expect(component.length).toBe(multipleOrders.length);
	// });
});
