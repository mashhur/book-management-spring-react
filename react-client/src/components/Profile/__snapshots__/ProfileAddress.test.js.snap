// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile address component test casses when an api call is made when request fails should match the error snapshot 1`] = `ReactWrapper {}`;

exports[`Profile address component test casses when an api call is made when request is success should match the success snapshot 1`] = `ReactWrapper {}`;

exports[`Profile address component test casses when an api call is made while loading should match the default loading snapshot 1`] = `ReactWrapper {}`;
